src/connector/src/changeable_fields.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// THIS FILE IS AUTO_GENERATED. DO NOT EDIT
+// UPDATE WITH: ./risedev generate-with-options
+
+use std::collections::{HashMap, HashSet};
+use std::sync::LazyLock;
+
+/// Map of source connector names to their changeable field names
+pub static SOURCE_CHANGEABLE_FIELDS: LazyLock<HashMap<String, HashSet<String>>> = LazyLock::new(|| {
+    let mut map = HashMap::new();
+    // KafkaProperties
+    map.insert(
+        "KafkaProperties".to_owned(),
```

Comment:
The key of the generated map is the struct type name. In actual usage, it seems hard to access the map from a source (for example, when checking kafka source, we need to access its entry by `"KafkaProperties"`). 

Therefore, shall we change to use the canonical type (obtained by std::any::type_name<T>) of property struct as the key? And then we can use the `type_name` of the corresponding `SourceProperties` of the source to access the map.

Reply:
I'm not sure about the final form, but better to restack #20780 on this PR to demonstrate how this is used

Reply:
> Therefore, shall we change to use the canonical type (obtained by std::any::type_name) of property struct as the key? And then we can use the type_name of the corresponding SourceProperties of the source to access the map.

I'm afraid it is not that easy.  

Originally, the proc_macro writes every possible fields of the top level Props for each connector to the yaml file. 
This approach uses two steps:
1) add more marks to the fields in the yaml
2) loads the yaml and pick the changeable fields

So at the time of code_gen, it does know the canonical type in the definition code.

I think the `XXXProperty` struct names are sufficient, we can check whether the keys(struct name) contain the desired connector name.

Reply:
> I'm not sure about the final form, but better to restack https://github.com/risingwavelabs/risingwave/pull/20780 on this PR to demonstrate how this is used

There is already a example field `sync_call_timeout`

you can see: 1) the changeable mark in the yaml (both source and sink);. 2) the field appears in the generated code, under the keys `KafkaProperties` for source and `KafkaConfig` for sink.